src/main/java/com/salaboy/conferences/agenda/controller/AgendaController.java

```diff
@@ -0,0 +1,86 @@
+package com.salaboy.conferences.agenda.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.salaboy.conferences.agenda.model.AgendaItem;
+import com.salaboy.conferences.agenda.model.Proposal;
+import com.salaboy.conferences.agenda.repository.AgendaItemRepository;
+import io.zeebe.client.api.response.ActivatedJob;
+import io.zeebe.client.api.worker.JobClient;
+import io.zeebe.spring.client.annotation.ZeebeWorker;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@Slf4j
+@RequestMapping
+public class AgendaController {
+
+    private final AgendaItemRepository agendaItemRepository;
+    private final ObjectMapper objectMapper;
+
+    public AgendaController(AgendaItemRepository agendaItemRepository, ObjectMapper objectMapper) {
+        this.agendaItemRepository = agendaItemRepository;
+        this.objectMapper = objectMapper;
+    }
+
+    @PostMapping
+    public Mono<String> newAgendaItem(@RequestBody AgendaItem agendaItem) {
+
+        log.info("> New Agenda Item Received: " + agendaItem);
+
+        if (agendaItem.getTitle().contains("fail")) {
+            throw new IllegalStateException("Something went wrong with adding the Agenda Item: " + agendaItem);
+        }
+
+        Mono<String> result = agendaItemRepository.save(agendaItem)
+                .doOnSuccess(i -> log.info("> Agenda Item Added to Agenda: {}", i))
+                .doOnError(i -> log.info("> Agenda Item NOT Added to Agenda: {}", i))
+                .map(i -> !Strings.isNullOrEmpty(i.getId()) ? "Agenda Item Added to Agenda" : "Agenda Item Not Added");
+
+        result.subscribe();
+
+        return result;
+    }
+
+    @GetMapping
+    public Flux<AgendaItem> getAll() {
+        return agendaItemRepository.findAll();
+    }
+
+    @GetMapping("/day/{day}")
+    public Mono<Set<AgendaItem>> getAllByDay(@PathVariable(value = "day", required = true) final String day) {
+        return agendaItemRepository.findAllByDay(day).collect(Collectors.toSet());
+    }
+
+    @GetMapping("/{id}")
+    public Mono<AgendaItem> getById(@PathVariable("id") String id) {
+
+        return agendaItemRepository.findById(id);
+    }
+
+    @DeleteMapping("/")
+    public Mono<Void> clearAgendaItems() {
+
+        log.info(">>> Deleting all");
+        return agendaItemRepository.deleteAll();
+    }
+
+    @ZeebeWorker(name = "agenda-worker", type = "agenda-publish")
+    public void newAgendaItemJob(final JobClient client, final ActivatedJob job) {
+        Proposal proposal = objectMapper.convertValue(job.getVariablesAsMap().get("proposal"), Proposal.class);
+        String[] days = {"Monday", "Tuesday"};
```

Comment:
@mcruzdev what do you think about extracting the time and day calculator to a separate method or a class that we can test independently?

Reply:
It is a good idea, thanks ;)